### DISCUSSION TITLE
How to add custom metric?

### DISCUSSION THREAD

**Discussion by ArMaxik** [open-mmlab/mmclassification#1323] [Q&A] [answered]
Hello!

I've implemeted my own Dataset class with overloaded `evaluate()` method, wich can calculate my custom metric. Then I've changed config:
```
dataset_type = 'MyDataset'
...
evaluation = dict(interval=1, metric=['f1_score', 'custom_metric', 'accuracy'], save_best='accuracy_top-1')
```

But during training custom metric has been calculated once, only after 1 epoch. Then it dissapeared and there are only `'f1_score', 'accuracy'` in the log.

What i should do to have my new metric works?

**Comment by Ezra-Yu** [comment, marked as answer]
please use [the 1.x branch](https://github.com/open-mmlab/mmclassification/tree/dev-1.x). In the 1.x branch, this issue has been resolved. 
Moreover, we are not going to spend a lot of effort on the master branch at the moment and will probably just fix some bugs later on, with all new features being developed on 1.x.
